front/lib/api/poke/plugins/data_sources/custom_crawler.ts

```diff
@@ -0,0 +1,61 @@
+import config from "@app/lib/api/config";
+import { createPlugin } from "@app/lib/api/poke/types";
+import logger from "@app/logger/logger";
+import { ConnectorsAPI, Err, Ok } from "@app/types";
+
+export const customCrawlerPlugin = createPlugin({
+  manifest: {
+    id: "custom-crawler",
+    name: "Custom Crawler",
+    description: "Setup custom crawler to webcrawler data source",
+    resourceTypes: ["data_sources"],
+    args: {
+      crawler: {
+        type: "enum",
+        label: "Crawler",
+        description: "Select a crawler",
+        values: ["firecrawl", "default"],
+      },
+    },
+  },
+  isApplicableTo: (_auth, dataSource) => {
+    if (!dataSource) {
+      return false;
+    }
+
+    return dataSource.connectorProvider === "webcrawler";
+  },
+  execute: async (_auth, dataSource, args) => {
```

Comment:
Seems very much like Rust 😛 
`nit`
```suggestion
  execute: async (_, dataSource, args) => {
```